src/main/java/com/github/webicitybrowser/webicity/renderer/backend/html/tags/TagAction.java

```diff
@@ -0,0 +1,36 @@
+package com.github.webicitybrowser.webicity.renderer.backend.html.tags;
+
+import com.github.webicitybrowser.spec.css.parser.CSSParser;
+import com.github.webicitybrowser.spec.css.parser.tokenizer.CSSTokenizer;
+import com.github.webicitybrowser.spec.css.parser.tokens.Token;
+import com.github.webicitybrowser.spec.css.rule.CSSRule;
+import com.github.webicitybrowser.spec.css.rule.CSSRuleList;
+import com.github.webicitybrowser.spec.dom.node.Element;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.HashMap;
+
+public interface TagAction {
+
+	CSSRuleList getCSSRuleList(Element element);
+
+	default CSSRuleList createRuleList(Reader childContextReader) {
```

Comment:
This would better be located in a different class, perhaps a static utility class.
The TagAction is meant for various tag triggers in general, not just styling-related stuff